irods/manager/data_object_manager.py

```diff
@@ -370,7 +385,7 @@ def make_FileOpenRequest(**extra_opts):
                 createMode=0,
                 openFlags=flags,
                 offset=0,
-                dataSize=-1,
+                dataSize=int(options.get(kw.DATA_SIZE_KW,'0')),
```

Comment:
We should send 0 for servers earlier than 4.3.1 due to irods/irods#7160. -1 is actually a useful concept now that it's not causing issues because it indicates that the file size is unknown (rather than 0, which is a valid file size).